user_tools/src/spark_rapids_tools/configuration/tools_config.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Container for the custom tools configurations. This is the parts of the configuration that can
+be passed as an input to the CLI"""
+
+import json
+from typing import Union, Optional
+
+from pydantic import BaseModel, Field, ValidationError
+from pydantic_core import PydanticCustomError
+
+from spark_rapids_tools import CspPathT
+from spark_rapids_tools.configuration.runtime_conf import ToolsRuntimeConfig
+from spark_rapids_tools.utils import AbstractPropContainer
+
+
+class ToolsConfigInfo(BaseModel):
+    """
+    High level metadata about the tools configurations (i.e., the api-version and any other relevant
+    metadata).
+    """
+    api_version: float = Field(
+        description='The version of the API that the tools are using. '
+                    'This is used to test the compatibility of the '
+                    'configuration file against the current tools release',
+        examples=['1.0'],
+        le=1.0,  # minimum version compatible with the current tools implementation
```

Comment:
does this automatically fail to read of its not 1.0?  does it fail the entire tools run?

Reply:
Yes, it will fail early in the phase in which we verify the arguments. As we move fwd, we can have a minimum compatible version. For example, backward compatibe with older formats can be expressed as `ge(1.1) and le(2.0)`
The point here is that the execution should not proceed because if the user is using out-daded configuration that does not fit the tools assumptions any more.